ts/components/messages/MessageAgenda.tsx

```diff
@@ -344,45 +400,70 @@ class MessageAgenda extends React.PureComponent<Props, State> {
     return null;
   }
 
-  private loadMoreData() {
-    // This state trigger progress bar
+  private isDeadlinesLoadingComplete = () => {
+    if (
+      this.getMessageToLoadFromSections(this.props.sections) ===
+        this.props.sectionToLoad &&
+      !this.state.isDeadlinesLoaded
+    ) {
+      this.setState({
+        isDeadlinesLoaded: true
+      });
+    }
+  };
+
+  private getMessageToLoadFromSections = (sections: Sections) => {
+    // tslint:disable-next-line: no-object-mutation
+    this.messageToLoadFromSections = 0;
+    sections.forEach(s => {
```

Comment:
```typescript
private getMessageToLoadFromSections = (sections: Sections) => {
    const messageToLoadFromSections = sections.reduce(
      (acc, curr) => acc + curr.data.length,
      0
    );
    const numMessagesToRender =
      messageToLoadFromSections + this.state.numMessagesToRender;
    this.setState({
      numMessagesToRender
    });
    return messageToLoadFromSections;
  };
```

You can avoid using `this.messageToLoadFromSections`. 
Anyway you are returning a value from a class method which is held in a class member